custom_components/zwave_mqtt/binary_sensor.py

```diff
@@ -21,26 +22,261 @@
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 4: Heat - Value Id's 1, 2, 5, 6 (heat/underheat)
+        NOTIFICATION_TYPE: NOTIFICATION_HEAT,
+        NOTIFICATION_VALUES: [1, 2, 5, 6],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_HEAT,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 4: Heat - All other Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_HEAT,
+        NOTIFICATION_VALUES: [3, 4, 8, 10, 11],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_HEAT,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 5: Water - Value Id's 1, 2, 3, 4
+        NOTIFICATION_TYPE: NOTIFICATION_WATER,
+        NOTIFICATION_VALUES: [1, 2, 3, 4],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_MOISTURE,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 5: Water - All other Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_WATER,
+        NOTIFICATION_VALUES: [5],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_MOISTURE,
+        NOTIFICATION_SENSOR_ENABLED: False,
+    },
+    {
+        # Index 6: Access Control - Value Id's 1, 2, 3, 4 (Lock)
+        NOTIFICATION_TYPE: NOTIFICATION_ACCESS_CONTROL,
+        NOTIFICATION_VALUES: [1, 2, 3, 4],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_LOCK,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 6: Access Control - Value Id 22 (door/window open)
+        NOTIFICATION_TYPE: NOTIFICATION_ACCESS_CONTROL,
+        NOTIFICATION_VALUES: [22],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_DOOR,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 1, 2 (intrusion)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [1, 2, 3, 4],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SAFETY,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 3, 4, 9 (tampering)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [3, 4, 9],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SAFETY,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 5, 6 (glass breakage)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [5, 6],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_SAFETY,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 7: Home Security - Value Id's 7, 8 (motion)
+        NOTIFICATION_TYPE: NOTIFICATION_HOME_SECURITY,
+        NOTIFICATION_VALUES: [7, 8],
+        NOTIFICATION_DEVICE_CLASS: DEVICE_CLASS_MOTION,
+        NOTIFICATION_SENSOR_ENABLED: True,
+    },
+    {
+        # Index 8: Power management - All Value Id's
+        NOTIFICATION_TYPE: NOTIFICATION_POWER_MANAGEMENT,
+        NOTIFICATION_VALUES: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15],
```

Comment:
Another comment, if these are individual binary sensors...

Battery power is actually a multi-state variable.

![image](https://user-images.githubusercontent.com/791794/80450907-6b6eef80-88d7-11ea-81a4-e3f4c2a64198.png)

You could represent them as individual binary sensors, but the same Clear logic would not apply, as these sensors could not all be on at the same time. Here is an illustration:

![image](https://user-images.githubusercontent.com/791794/80450863-51cda800-88d7-11ea-8761-fd83de78222f.png)

The battery level status could not be both "fully charged" and "charge battery now" at the same time. You'd either need a normal sensor, represented with text states (yuck?), or individual binary sensors that are mutually exclusively on.

Reply:
I have never seen a device using this and thus impossible for me to test nor code.
If you give me an example how this looks like in the MQTT messages I'd be happy to split them up. For now I consider this as an edge case and ignore it.

Reply:
Well, I don't have a device with these properties, but if OZW were following the spec then the MQTT messages would arrive in order of the timing diagram.
1. mqtt sends 0x0D - HA sets "battery is fully charged" to On
2. mqtt sends 0x00 - HA sets all power sensors to Off
3. mqtt sends 0x0e - HA sets "charge battery soon" sensor to On
4. mqtt sends 0x0f - HA sets "charge battery now" sensor to On
5. mqtt sends 0x0f - no state change in HA
6. mqtt sends 0x00 - HA sets all power sensors to Off
7. mqtt sends 0x0d - HA sets "battery is fully charged" to On

After step 4, both "charge battery soon" and "charge battery now" are On, which is not quite right.

Reply:
You are correct but

1) I still have no idea how this is presented in the mqtt data structure, will there be another list in the value for example ?

2) I still think this is an edge case as I have never seen any devices with these values. My focus was on the values that can be easily presented as a binary sensor. The fallback is in the sensor with the raw sensor data. As soon as the first user pops up with this need we can extend the code but adding it in now feels beyond MVP scope.

What I will do is add some comments in the code for and ignore the values that can no be properly translated into a boolean

Reply:
They are just [List values](https://github.com/OpenZWave/open-zwave/blob/5220be772970b922fd6304f9d347d025a6511e7a/config/NotificationCCTypes.xml#L306-L313) like all of the others.
```xml
		<AlarmEvent id="12" name="Battery Charging">
		</AlarmEvent>
		<AlarmEvent id="13" name="Battery Charged">
		</AlarmEvent>
		<AlarmEvent id="14" name="Battery Low">
		</AlarmEvent>
		<AlarmEvent id="15" name="Battery Critical">
		</AlarmEvent>
```
I would expect that to translate to (this condensed value):
```json
{
    "Label": "Power Management",
    "Value": {
        "List": [
            {
                "Value": 0,
                "Label": "Clear"
            },
            {
                "Value": 12,
                "Label": "Battery Charging"
            }
            {
                "Value": 13,
                "Label": "Battery Charged"
            }
            ...
        ],
        "Selected": "Battery Charged",
        "Selected_id": 13
    },
    "Type": "List",
    "CommandClass": "COMMAND_CLASS_NOTIFICATION",
    "Index": 8,
}
```
The only difference is that each of these represent a value for a single variable, so they cannot all be true at the same time. Sure, you can represent them each as binary sensors, but only one can be on at a time.

My main point was that this particular set of values is a bit more complicated than the others.

Reply:
Sorry, I mixed up my comments :-) I was talking about the case where the value itself has another list underneath. In this case it's just a regular list indeed. 

@Fishwaldo when the state moved from "charging" to "charged" will there be a "clear" event in between ?

Reply:
Already solved.